src/index.ts

```diff
@@ -43,19 +64,37 @@ export class Amplitude {
    * (whichever comes first), as well as on app close.
    *
    * @param eventType The name of the event you wish to track.
+   * @param eventProperties The event's properties.
+   * @param extra Extra untyped parameters for use in middleware.
    */
   logEvent(
     eventType: string,
     eventProperties?: Record<string, unknown>,
+    extra?: MiddlewareExtra,
   ): Promise<boolean> {
-    if (eventProperties && Object.keys(eventProperties).length > 0) {
-      return AmplitudeReactNative.logEventWithProperties(
-        this.instanceName,
-        eventType,
-        eventProperties,
-      );
-    }
-    return AmplitudeReactNative.logEvent(this.instanceName, eventType);
+    return AmplitudeReactNative.getUserId(this.instanceName).then((userId) =>
+      AmplitudeReactNative.getDeviceId(this.instanceName).then((deviceId) => {
+        const properties = eventProperties ?? {};
+        const event = {
+          event_type: eventType,
+          user_id: userId,
+          device_id: deviceId,
+          event_properties: properties,
+        };
+        if (!this._runMiddlewares(event, extra)) {
+          return Promise.resolve(false);
+        }
+
+        if (Object.keys(properties).length > 0) {
+          return AmplitudeReactNative.logEventWithProperties(
+            this.instanceName,
+            eventType,
```

Comment:
We should use the modified event values here. `event.eventType`, `event.eventProperties`

Reply:
fixed